### PR TITLE
Add Security Rule Options

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

- N/A

## Description of Changes

- Adds Security Rule Categories
- Adds Security Rules (without conditions, just names and descriptions)

### CODE REVIEW

File: database/src/migrations/20240808000000_biohub_security_rules.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`
+   ---------------------------------------------------
+   ----- inserting reasons into category table -------
+   ---------------------------------------------------
+    INSERT INTO security_category(name, description, record_Effective_date)
+    VALUES 
+    ('Government Interests', 'Government Interests', now()),
```
Comment (mauberti-bc) [inline database/src/migrations/20240808000000_biohub_security_rules.ts:18]: Change to "Government Interest", singular

File: database/src/migrations/20240808000000_biohub_security_rules.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Proprietary'), 'Time-Limited Restriction', 'Proprietary due to Time-Limited Restriction'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Proprietary'), 'First Nations Land', 'Proprietary due to First Nations Land'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Statutory Constraints'), 'Provincial Statute', 'Secured due to Provincial Statute'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Statutory Constraints'), 'Federal Statute', 'Secured due to Federal Statute'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Mule Deer Telemetry', 'Secured due to Mule Deer Telemetry'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Mule Deer Survey Data', 'Secured due to Mule Deer Survey Data'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Modeled Freshwater Fish Habitat Capability Mapping', 'Secured due to Experimental Technology'),
```
Comment (mauberti-bc) [inline database/src/migrations/20240808000000_biohub_security_rules.ts]: I don't think we need such a specific rule (modeled freshwater fish habitat capability mapping). Did you see this somewhere?

File: database/src/migrations/20240808000000_biohub_security_rules.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Predator Reduction', 'Secured due to Predator Reduction Information'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Bighorn or Thinhorn Sheep', 'Secured due to Species of Sheep'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Rocky Mountain Elk Telemetry', 'Secured due to Rocky Mountain Elk Telemetry'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Rocky Mountain Elk  Survey Data', 'Secured due to Rocky Mountain Elk Survey Data'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Harvested Species', 'Secured due to Increased Risk of Illegal Harvest'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Secured Caribou Herd', 'Secured due to association with a Secured Caribou Herd'),
```
Comment (mauberti-bc) [inline database/src/migrations/20240808000000_biohub_security_rules.ts]: It seems redundant to include "Secured" in the name - let's simplify the names by just removing "Secured"

**Comment by mauberti-bc** [comment]
@AMEIJER1 Let me know what you think: it seems complex to have separate telemetry/data/dna/etc. rules for each species. It will be *much* easier to manage security if we just have one rule per species/category, like: Caribou government interest, caribou persecution and harm, grizzly bear persecution and harm, rattlesnake persecution and harm. If we ever need to remove rules after a certain amount of time, we can do things like: remove "caribou" rule where data type = telemetry. A bit of a 180 on our last conversation but I don't think the highly specific rules are sustainable.

File: database/src/migrations/20240808000000_biohub_security_rules.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Secured Caribou Telemetry', 'Secured due to Caribou Telemetry'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Secured Moose Data', 'Secured Moose Data'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Secured Moose Telemetry', 'Secured due to Moose Telemetry'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Moose Stratification', 'Secured due to Moose Stratification'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Grizzly Bear DNA', 'Secured due to Grizzly Bear DNA'),
+    (now(), (SELECT security_category_id FROM security_category WHERE name = 'Government Interests'), 'Grizzly Bear Telemetry', 'Secured due to Grizzly Bear Telemetry'),
```
Comment (mauberti-bc) [inline database/src/migrations/20240808000000_biohub_security_rules.ts]: Collapse into one grizzly government interest rule

**Review by NickPhura — APPROVED** [review]
LGTM

**Pull request merged**